### PR TITLE
Adds redesigned UI for SleepTimerFragment

### LINKED ISSUE #145: Add sleep timer presets
### Add preset times
As a user, I would like to have some quick presets for the sleep timer as it currently takes too much effort to set the timer. For example, there could be buttons for:
* 1 hour 
* 2 Hours
* 4 Hours 
* 8 Hours

### Presets as the default - Time selector as advanced option
There should be a number of presets instead of the current radial dials (which can be confusing for tired people). Allow the user to select a specific time like it is now BUT present pre-sets before making the user have to hop through the loops of that radial dial. 

I guess the real point of this request is that the radial dial is not very nice. Image you are going to bed and don't have your glasses on... it's really hard to set that dial.

### PR DESCRIPTION
**Pull request by ashutoshgngwr**
### Changes
- Redesign the sleep scheduling interface

screenshot

### Testing
- [x] Tested on a physical device
- [x] Added or modified unit test cases

### Others
- Fixes #145 
- Connects #0

### CODE REVIEW

**Comment by ashutoshgngwr** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ashutoshgngwr/noice/pullRequest?prid=5434825&bid=17822895)

File: app/src/main/java/com/github/ashutoshgngwr/noice/widget/CountdownTextView.kt [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+  constructor(@NonNull context: Context) : super(context)
+  constructor(@NonNull context: Context, @Nullable attrs: AttributeSet) : super(context, attrs)
+  constructor(@NonNull context: Context, @Nullable attrs: AttributeSet, defStyleAttr: Int) : super(
+    context,
+    attrs,
+    defStyleAttr
+  )
+
+  /**
```
Comment (ashutoshgngwr) [inline app/src/main/java/com/github/ashutoshgngwr/noice/widget/CountdownTextView.kt:40]: Codacy Issue found: [CommentOverPrivateProperty - [countdownUntilMillis] at /src/app/src/main/java/com/github/ashutoshgngwr/noice/widget/CountdownTextView.kt:40:3](https://app.codacy.com/manual/ashutoshgngwr/noice/pullRequest?prid=5434825)

**Comment by aproposnix** [comment]
I think having to set an alarm, manually, every night you use it, a little overkill. Provide the user with some common schedules (1h, 2h, 4h, etc) in one click and the custom schedule for those who would like to be more granular.

**Comment by ashutoshgngwr** [comment]
Technically 1h and 4h is one click while 2h is 2 clicks?

**Comment by ashutoshgngwr** [comment]
@aproposnix I am moving ahead with this. While this may not fully address your concern for presets, it certainly does make scheduling auto-sleep a lot easier. I think this is most safe option moving forward while avoiding unnecessary clutter in the app.

**Comment by aproposnix** [comment]
OK.. now using and loving this. Thanks for this. It's MUCH easier now.

**Pull request merged**